source/grid/manifold_lib.cc

```diff
@@ -446,33 +597,18 @@ get_new_point (const ArrayView<const Point<spacedim>> &vertices,
         // rarely have more than 9 points)
         Tensor<1,3> direction_3d;
         for (unsigned int c=0; c<spacedim; ++c)
-          direction_3d[c] = direction[c];
-        bool found = false;
-        for (unsigned int j=0; j<directions.size(); ++j)
-          if ((directions[j]-direction_3d).norm_square() < 1e-28)
-            {
-              merged_weights[j] += weights[i];
-              found = true;
-              break;
-            }
-        if (found == false)
-          {
-            directions.push_back(direction_3d);
-            merged_weights.push_back(weights[i]);
-          }
+          directions_3d[i][c] = directions[i][c] / distances[i];
```

Comment:
Do we still need to copy the data into a new array? Can we just use the original one? And can't we have `directions[j]` already normalization, i.e., on the unit sphere? (I might just have gotten lost by all the normalization steps we already do in this class.)

Reply:
The comment above seems to be outdated. The weights are already merged at this point.

Reply:
Regarding @kronbichler's comment: We already normalize in line 389, so we can turn this normalization (or the one in line 586) in an `Assert`.